libs/defi/shared/src/components/TxButton/hooks/useApprovalButton.tsx

```diff
@@ -103,104 +110,58 @@ export const useApprovalButton = (args: UseApprovalButtonProps) => {
   });
 
   const onWrite = useCallback(() => {
-    const act = {
-      type: 'approval',
-      title: intl.formatMessage(
-        { defaultMessage: 'Approve {token}' },
-        { token: args.token.symbol },
-      ),
-      subtitle: intl.formatMessage(
-        {
-          defaultMessage: 'Approving {token}',
-        },
-        {
-          token: args.token.symbol,
-        },
-      ),
-      endIcon: (
-        <BadgeIcon
-          badgeContent={
-            <FaCircleCheckRegular color="primary" sx={{ fontSize: 10 }} />
-          }
-        >
-          <TokenIcon token={args.token} sx={{ fontSize: 36 }} />
-        </BadgeIcon>
-      ),
+    const updated = pushActivity({
+      ...activity,
       status: 'pending',
-      tokenIn: args.token,
-      amountIn: args.amount,
-    } as const;
-    if (!args.disableActivity) {
-      const activity = pushActivity(act);
-      setAct(activity);
-    } else {
-      setAct({
-        id: Date.now().toString(),
-        createdOn: Date.now(),
-        ...act,
-      });
-    }
-    args.callbacks?.onWrite?.();
-  }, [
-    args.amount,
-    args.callbacks,
-    args.disableActivity,
-    args.token,
-    intl,
-    pushActivity,
-  ]);
+      title: activityOptions.approval.title(activity, intl),
+      subtitle: activityOptions.approval.subtitle(activity, intl),
+      icon: activityOptions.approval.icon(activity),
+    });
+    setActivity(updated);
```

Comment:
The `useActivity` hook I had made removes the need for us to track activity locally. Did you not like its complexity?

Reply:
oh, I think I simply overlooked and not used it. It is extracting this to a custom hook, right?

Reply:
Yeah it just removed the need to track your activity id. And instead of using 3 hooks you get it all in one. Usually whatever uses these uses them all.

Reply:
ok, I updated, it indeed looks cleaner!